crates/re_renderer/src/renderer/depth_cloud.rs

```diff
@@ -242,12 +242,36 @@ fn create_and_upload_texture<T: bytemuck::Pod>(
     let width_blocks = depth_cloud.depth_dimensions.x / format_info.block_dimensions.0 as u32;
     let bytes_per_row_unaligned = width_blocks * format_info.block_size as u32;
 
+    // TODO(andreas): CpuGpuWriteBelt should make it easier to do this.
+    let bytes_per_row_aligned =
+        wgpu::util::align_to(bytes_per_row_unaligned, wgpu::COPY_BYTES_PER_ROW_ALIGNMENT);
+
     let mut depth_texture_staging = ctx.cpu_write_gpu_read_belt.lock().allocate::<T>(
         &ctx.device,
         &ctx.gpu_resources.buffers,
         data.len(),
     );
-    depth_texture_staging.extend_from_slice(data);
+
+    // Fill with a single copy if possible, otherwise do multiple, filling in padding.
+    if bytes_per_row_aligned == bytes_per_row_unaligned {
+        depth_texture_staging.extend_from_slice(data);
+    } else {
+        let num_padding_bytes = (bytes_per_row_aligned - bytes_per_row_unaligned) as usize;
+        // Sanity check the padding size. If this happens something is seriously wrong, as it would imply
+        // that we can't express the required alignment with the block size.
+        debug_assert!(
+            num_padding_bytes % std::mem::size_of::<T>() == 0,
+            "Padding is not a multiple of pixel size. Can't correctly pad the texture data"
+        );
+        let padding = std::iter::repeat(T::zeroed())
+            .take(num_padding_bytes / std::mem::size_of::<T>())
+            .collect::<Vec<_>>();
+
+        for row in data.chunks(width_blocks as usize) {
```

Comment:
was pondering just now if this is really correct. But yeah `T` is a block, so the question is how many blocks this is wide!